### PR TITLE
ref(apple): dSYMs restructuring

### PR DESCRIPTION
**Pull request by philipphofmann**
This PR restructures the dSYMs page for Apple to make it easier to understand, moves the Bitcode related section down, as Bitcode is now deprecated with Xcode 14, and mentions how to upload source bundles for source context to work.

Fixes GH-5823

As I tried to add the information on how to make source context work, I would have had to add the info in many different places. Instead, I restructured the page quite a bit to simplify it, and now I have to add the info in fewer places.

### CODE REVIEW

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 1]
```diff
@@ -223,3 +116,63 @@ export SENTRY_URL=https://mysentry.invalid/
...
+
+If [Bitcode](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/AppThinning/AppThinning.html#//apple_ref/doc/uid/TP40012582-CH35-SW2), no longer available [starting with Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes), **is enabled** in your project, you need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry.
+You can achieve this by using either:
+
+- Fastlane's [download_dsyms](https://docs.fastlane.tools/actions/download_dsyms/) action and then upload the dSYMs with the [Sentry Fastlane plugin](#fastlane).
+- [Sentry App Store Connect integration](#appstore-connect), which fetches the dSYMS direclty from App Store Connect.
+- Manually from App Store Connect:
+  - Open Xcode Organizer, go to your app, and click _Download dSYMs..._
+  - Login to App Store Connect, go to your app, go to “Activity", click the build number to go into the detail page, and click _Download dSYM_.
+  - Manually upload the dSYMs with [sentry-cli](#sentry-cli).
```
Comment (lizokm) [inline src/platform-includes/debug-symbols-apple/_default.mdx:176]:
```suggestion
- Download Fastlane's [download_dsyms](https://docs.fastlane.tools/actions/download_dsyms/) action and then upload the dSYMs with the [Sentry Fastlane plugin](#fastlane).
- Use the [Sentry App Store Connect integration](#appstore-connect), which fetches the dSYMS directly from App Store Connect.
- Download manually from App Store Connect:
  - Open Xcode Organizer, go to your app, and click _Download dSYMs..._
  - Log in to App Store Connect, go to your app > “Activity".
  - Click the build number to go into the detail page, and click _Download dSYM_.
  - Manually upload the dSYMs with [sentry-cli](#sentry-cli).
```

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 2]
```diff
@@ -223,3 +116,63 @@ export SENTRY_URL=https://mysentry.invalid/
...
+
+These dSYMs contain obfuscated symbol names and paths. You must upload the BCSymbolMap file stored in your xcarchive so Sentry can  properly symbolicate your crash reports. You can do this by using the upload-dif command of either [sentry-cli](#sentry-cli) or the [Sentry Fastlane plugin](#fastlane), which will automatically find and upload all BCSymbolMap files, given the path to the app's xcarchive.
```
Comment (lizokm) [inline src/platform-includes/debug-symbols-apple/_default.mdx:178]:
```suggestion
These dSYMs contain obfuscated symbol names and paths. You must upload the BCSymbolMap file stored in your xcarchive so Sentry can properly symbolicate your crash reports. You can do this by using the upload-dif command of either [sentry-cli](#sentry-cli) or the [Sentry Fastlane plugin](#fastlane), which will automatically find and upload all BCSymbolMap files once it has the path to the app's xcarchive.
```
Want to make sure that the update I made to the last line makes sense? Am I understanding what you meant by "given the path to the app's...?" correctly?

**Review by lizokm — COMMENTED** [review]
Thanks so much for this cleanup! I've made some copy tweaks, feel free to reach out if you want to talk about any of them.

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 3]
```diff
@@ -1,153 +1,47 @@
...
-##### `Issue Details`
-
-![Warning to update credentials in an issue's details](issue-details-warning.png)
-
-##### `Project Settings`
+## sentry-cli
 
-![Warning to update credentials in sidebar](sidebar-warning.png)
+Use the `sentry-cli upload-dif` command to upload dSYMs to Sentry. The command will recursively scan the provided folders or ZIP archives. Files that have already been upload are skipped automatically.
+We recommend setting the `--include_sources` option to include sources for <PlatformLink to="/data-management/debug-files/source-context/">source context</PlatformLink> to work.
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply src/platform-includes/debug-symbols-apple/_default.mdx:22]: When using the sentry-cli you only have to add `--include_sources` , when using Fastlane you have to set `include_sources` to `true`, if you set it to `false` it doesn't work.

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 4]
```diff
@@ -160,14 +54,13 @@ api_host: 'https://mysentry.invalid/'
...
+## Xcode Build Phase
 
-You need to have an auth token for this to work. You can [create an auth token here](https://sentry.io/api/).
+Your project's dSYM can be uploaded during Xcode's build phase as a _Run Script_. For this you need to set the _DEBUG_INFORMATION_FORMAT_ to be _DWARF with dSYM File_. By default, an Xcode project will only have _DEBUG_INFORMATION_FORMAT_ set to _DWARF with dSYM File_ in _Release_ so make sure everything is set in your build settings properly. To upload the debug symbols, you need to do the following:
 
 1.  Download and install [sentry-cli](/product/cli/installation/)
-2.  You will need to copy the script below into a new _Run Script_ and set your _AUTH_TOKEN_, _ORG_SLUG_, and _PROJECT_SLUG_
+2.  You will need to copy the script below into a new _Run Script_ and set your _AUTH_TOKEN_, _ORG_SLUG_, and _PROJECT_SLUG_.
+3.  We recommend setting the `--include_sources` option to include sources for <PlatformLink to="/data-management/debug-files/source-context/">source context</PlatformLink> to work. Simply change `sentry-cli upload-dif` to `sentry-cli upload-dif --include_sources` in the copied script.
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply src/platform-includes/debug-symbols-apple/_default.mdx:66]: > For source context to work, we recommend setting the include_sources parameter to true:"

No, it should state only adding `--include_sources`.

File: src/platform-includes/debug-symbols-apple/_default.mdx (hunk 2 above)
Comment (philipphofmann) [reply src/platform-includes/debug-symbols-apple/_default.mdx:180]: I tried it differently. Please have another look.

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 5]
```diff
@@ -223,3 +116,63 @@ export SENTRY_URL=https://mysentry.invalid/
...
+#### `Custom Repositories`
+
+![Warning to update credentials in Custom Repositories](custom-repositories-warning.png)
+
+When using Fastlane's _upload_to_testflight_ action, it is recommended to pass the _skip_waiting_for_build_processing_ parameter to speed up the action.
+
+## Bitcode {#dsym-with-bitcode}
+
+If [Bitcode](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/AppThinning/AppThinning.html#//apple_ref/doc/uid/TP40012582-CH35-SW2), no longer available [starting with Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes), **is enabled** in your project, you need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry.
+You can achieve this by using either:
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply src/platform-includes/debug-symbols-apple/_default.mdx:171]: Almost, I changed the second Xcode 14 to Xcode 14.

**Comment by philipphofmann** [comment]
Thank you, @lizokm, for the great review 😀.

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 6]
```diff
@@ -223,3 +120,61 @@ export SENTRY_URL=https://mysentry.invalid/
...
+
+#### `Custom Repositories`
+
+![Warning to update credentials in Custom Repositories](custom-repositories-warning.png)
+
+When using Fastlane's _upload_to_testflight_ action, it's recommended to pass the _skip_waiting_for_build_processing_ parameter to speed up the action.
+
+## Bitcode {#dsym-with-bitcode}
+
+The App Store will not accept bitcode submissions from [Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes). If you’re using Xcode 13, you’ll need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry using one of the methods below:
```
Comment (lizokm) [inline src/platform-includes/debug-symbols-apple/_default.mdx:170]:
```suggestion
The App Store will not accept bitcode submissions from [Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes). If you’re using Xcode 14, you’ll need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry using one of the methods below:
```

Should it be "if you're using Xcode 14?"

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 7]
```diff
@@ -223,3 +120,61 @@ export SENTRY_URL=https://mysentry.invalid/
 ```
 
 </Alert>
+
+## App Store Connect {#appstore-connect}
```
Comment (lizokm) [inline src/platform-includes/debug-symbols-apple/_default.mdx:124]:
```suggestion
## App Store Connect {#appstore-connect} Integration
```

File: src/platform-includes/debug-symbols-apple/_default.mdx [hunk 8]
```diff
@@ -223,3 +120,61 @@ export SENTRY_URL=https://mysentry.invalid/
...
+## Bitcode {#dsym-with-bitcode}
+
+The App Store will not accept bitcode submissions from [Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes). If you’re using Xcode 13, you’ll need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry using one of the methods below:
+
+- Download Fastlane's [download_dsyms](https://docs.fastlane.tools/actions/download_dsyms/) action and then upload the dSYMs with the [Sentry Fastlane plugin](#fastlane).
+- Use the [Sentry App Store Connect integration](#appstore-connect), which fetches the dSYMS directly from App Store Connect.
+- Download manually from App Store Connect:
+  - Open Xcode Organizer, go to your app, and click _Download dSYMs..._
+  - Log in to App Store Connect, go to your app > “Activity".
+  - Click the build number to go into the detail page, and click _Download dSYM_.
```
Comment (lizokm) [inline src/platform-includes/debug-symbols-apple/_default.mdx:177]:
```suggestion
  - Click the build number to go into the "detail" page, and click _Download dSYM_.
```

**Review by lizokm — APPROVED** [review]
I think this looks great! Just left a few tiny comments.

File: src/platform-includes/debug-symbols-apple/_default.mdx (hunk 6 above)
Comment (philipphofmann) [reply src/platform-includes/debug-symbols-apple/_default.mdx:170]: No, it should be 13. If you use Xcode 14, it doesn't work.

**Pull request merged**